gui/src/app/components/TabWidget.tsx

```diff
@@ -29,7 +40,12 @@ const TabWidget: FunctionComponent<TabWidgetProps> = ({ labels, children }) => {
       </Tabs>
       <Box flex="1" overflow="scroll">
         {children.map((child, i) => (
-          <CustomTabPanel key={i} value={index} index={i}>
+          <CustomTabPanel
+            key={i}
+            value={index}
+            index={i}
+            tabsThatHaveBeenViewed={tabsThatHaveBeenViewed}
```

Comment:
(Assuming the above change to boolean[] is done) I think it would be clearer if this is just a `mounted : Boolean`, rather than the child knowing about this entire array